src/content/install-dialog.js

```diff
@@ -33,6 +33,25 @@ gDownloader.addProgressListener(() => {
 /****************************** DETAIL DISPLAY *******************************/
 
 gDownloader.scriptDetails.then(scriptDetails => {
+
+  return new Promise((resolve, reject) => {
+    chrome.runtime.sendMessage({
+        'name': 'UserScriptGetByName',
+        'scriptNamespace' : scriptDetails.namespace,
+        'scriptName' : scriptDetails.name
+      }, (installedScriptDetails) => {
+        if (chrome.runtime.lastError) {
+          console.error(chrome.runtime.lastError);
+          reject(chrome.runtime.lastError);
```

Comment:
This reject is never going to get caught. There's no `catch` in the promise chain. Also, depending on the error it may be considered non-fatal and thus there's no need to stop the entire chain.

What you can do instead is remove line 44 and then on line 53 (number 52 after removing 44) add a `.catch` that logs the error, performs any additional checks, such as if the error is fatal or not, and if the error is not fatal then `return scriptDetails` or if it is fatal `throw [new]error`.

Reply:
How do I know if the error is fatal?

Reply:
Good question. And for that I don't have a good answer. For this particular case the majority of what you might receive will likely be not fatal. Think of it this way, "If we fail to get the current version of a script is that a roadblock for displaying the rest of the install dialog?" For most situations I'd say no. However, if there are errors that we can handle or otherwise should display to the user (usually these are errors that we generate) then they should be handled appropriately before `return scriptDetails`.

Unfortunately it can be a major guessing game trying to figure out what Mozilla's APIs may throw at us and whether we can handle them or not.

Reply:
How do you continue with .then after .catch? I can't get this to work.

Reply:
Here's a gist. The main point is where you attach the `.catch`. You need to attach it such that you still have a reference to `scriptDetails`.
https://gist.github.com/Sxderp/6d37f4e42a02ec4f7f8962f4ef5579f4